From dd344ee726aec9f0b08aa5248d2af068d41def95 Mon Sep 17 00:00:00 2001
From: Vincent Abriou <vincent.abriou@st.com>
Date: Fri, 17 Jun 2016 10:47:40 +0200
Subject: [PATCH 3/6] mali: replace PAGE_CACHE_* and page_cache_{get,release}
 that have been dropped

Since kernel 4.6-rc3 PAGE_CACHE_SHIFT and page_cache_release have
been dropped. Use PAGE_CACHE and put_page instead.

Signed-off-by: Vincent Abriou <vincent.abriou@st.com>
---
 src/devicedrv/mali/linux/mali_memory_swap_alloc.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/devicedrv/mali/linux/mali_memory_swap_alloc.c b/src/devicedrv/mali/linux/mali_memory_swap_alloc.c
index 4793328..068e4d0 100755
--- a/src/devicedrv/mali/linux/mali_memory_swap_alloc.c
+++ b/src/devicedrv/mali/linux/mali_memory_swap_alloc.c
@@ -52,7 +52,7 @@ extern struct mali_mem_os_allocator mali_mem_os_allocator;
 #define MALI_SWAP_LOW_MEM_DEFAULT_VALUE (60*1024*1024)
 #define MALI_SWAP_INVALIDATE_MALI_ADDRESS (0)               /* Used to mark the given memory cookie is invalidate. */
 #define MALI_SWAP_GLOBAL_SWAP_FILE_SIZE (0xFFFFFFFF)
-#define MALI_SWAP_GLOBAL_SWAP_FILE_INDEX ((MALI_SWAP_GLOBAL_SWAP_FILE_SIZE) >> PAGE_CACHE_SHIFT)
+#define MALI_SWAP_GLOBAL_SWAP_FILE_INDEX ((MALI_SWAP_GLOBAL_SWAP_FILE_SIZE) >> PAGE_SHIFT)
 #define MALI_SWAP_GLOBAL_SWAP_FILE_INDEX_RESERVE (1 << 15) /* Reserved for CoW nonlinear swap backend memory, the space size is 128MB. */
 
 unsigned int mali_mem_swap_out_threshold_value = MALI_SWAP_LOW_MEM_DEFAULT_VALUE;
@@ -183,7 +183,7 @@ static void mali_mem_swap_out_page_node(mali_page_node *page_node)
 	dma_unmap_page(&mali_platform_device->dev, page_node->swap_it->dma_addr,
 		       _MALI_OSK_MALI_PAGE_SIZE, DMA_TO_DEVICE);
 	set_page_dirty(page_node->swap_it->page);
-	page_cache_release(page_node->swap_it->page);
+	put_page(page_node->swap_it->page);
 }
 
 void mali_mem_swap_unlock_single_mem_backend(mali_mem_backend *mem_bkend)
-- 
2.7.4

